addons/mail/static/src/models/web_client_view_attachment_view.js

```diff
@@ -40,10 +40,14 @@ registerModel({
             if (
                 this.thread.areAttachmentsLoaded &&
                 !this.thread.isLoadingAttachments &&
-                !this.thread.mainAttachment &&
                 this.thread.attachmentsInWebClientView.length > 0
             ) {
-                this.setMainAttachmentFromIndex(0);
+                if (!this.thread.mainAttachment) {
+                    this.setMainAttachmentFromIndex(0);
+                }
+                else if (!this.thread.mainAttachment?.isViewable) {
+                    this.thread.update({ mainAttachment: this.thread.attachmentsInWebClientView[0] });
+                }
```

Comment:
There's necessarily a mainAttachment (else if) but there isn't necessarily a "web client view" one
```suggestion
                else if (!this.thread.mainAttachment.isViewable && this.thread.attachmentsInWebClientView?.length) {
                    this.thread.update({ mainAttachment: this.thread.attachmentsInWebClientView[0] });
                }
```

Reply:
Thanks for your feedback!
Error scenario:
1. A user uploads a file that doesn’t support preview (e.g., .html).
2. Then, the user uploads a file that does support preview (e.g., .pdf).
Expected behavior:
* The system should preview the .pdf file.
Actual result:
* The system previews the .html file instead.
Cause and temporary fix:
* After calling this.setMainAttachmentFromIndex(0);, the _onChangeThreadAttachmentsInWebClientView method is triggered again, but the mainAttachment on the client side is not updated correctly — it still points to the .html file.
* Meanwhile, in the database, the mainAttachment has already been updated to the .pdf file.
* I haven’t identified the root cause of why the mainAttachment on the client isn't updated, but the current code works as a temporary fix.
* If it’s necessary to investigate further, I’ll continue to look into the root cause.

Reply:
I found the root cause: the fetchData function is triggered immediately after setMainAttachment, while the data hasn't been saved to the database yet — as a result, the fetched mainAttachment and the updated value are still the old one, which leads to the error.
Therefore, to resolve the issue, I added a flag to determine whether the fetch is happening during setMainAttachment. If so, it will not update mainAttachment, since the data fetched for mainAttachment has already been set by the setMainAttachment function.